Cache.js

```diff
@@ -0,0 +1,39 @@
+const redis = require('redis');
+const defaultHost = '127.0.0.1';
+const defaultPort = 6379;
+const timeToLiveInSeconds = 60;
+
+class Cache {
+  constructor ({ host = defaultHost, port = defaultPort, logger }) {
+    this.client = redis.createClient({ host, port });
+    this.logger = logger;
+    this.client.on('error', (error) => {
+      this.logger.error(error);
+    });
+
+    this.client.on('ready', () => {
+      this.logger.info(`Connected to redis at ${host}:${port}`);
+    });
+  }
+
+  /**
+   * @param {string} realtimeEventId - event from websocket stream
+   * @param {string} updatedAt - timestamp, avoids ignoring events with same id
+   * @param {string} type - one of botAction, botData, event
+   * @returns {Promise}
+   */
+  hasBeenConsumed(realtimeEventId, updatedAt, type) {
+    return new Promise((resolve) => {
+      const key = `${type}${realtimeEventId}${updatedAt}`;
+      this.client.set(key, '1', 'NX', 'ex', timeToLiveInSeconds, (err, res) => {
```

Comment:
what is '1', 'NX', 'ex' can we create more meaningful names here?

Reply:
'1' is just the value that is being set for the key.
'NX' is a flag that indicates that it should only set if the kv pair does not exist (NX). 
'ex' is the param for 'expires' 

Will refactor them into variables